src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,96 @@
 package core.basesyntax;
 
 public class SalaryInfo {
+    private static final int RECORD_DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOURS_STR_INDEX = 2;
+    private static final int RATE_STR_INDEX = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        return generateReport(names, data, dateFrom, dateTo);
+    }
+
+    private String generateReport(String[] names, String[] data, String dateFrom, String dateTo) {
+        StringBuilder reportBuilder = new StringBuilder();
+
+        initializeReportHeader(reportBuilder, dateFrom, dateTo);
+        int[] earnings = calculateEarnings(names, data, dateFrom, dateTo);
+        finalizeReport(reportBuilder, names, earnings);
+
+        return reportBuilder.toString();
+    }
+
+    private void initializeReportHeader(StringBuilder reportBuilder,
+                                        String dateFrom,
+                                        String dateTo) {
+        reportBuilder.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append(System.lineSeparator());
+    }
+
+    private int[] calculateEarnings(String[] names, String[] data, String dateFrom, String dateTo) {
+
+        int[] earnings = new int[names.length];
+
+        for (String record : data) {
+            String[] parts = record.split(" ");
+            if (parts.length == 4) {
+                String recordDate = parts[RECORD_DATE_INDEX];
+                String name = parts[NAME_INDEX];
+                String hoursStr = parts[HOURS_STR_INDEX];
+                String rateStr = parts[RATE_STR_INDEX];
+
+                for (int i = 0; i < names.length; i++) {
+                    if (name.equals(names[i]) && isWithinDateRange(recordDate, dateFrom, dateTo)) {
+                        earnings[i] += parseAndCalculateEarnings(hoursStr, rateStr);
+                    }
+                }
+            }
+        }
+        return earnings;
+    }
+
+    private boolean isWithinDateRange(String recordDate, String dateFrom, String dateTo) {
```

Comment:
This is still overcomplicated.
1. You can parse dateFrom and dateTo into `LocalDate` at the beginning of the calculations
2. Then, you can parse recordDate to `LocalDate` as well and pass all 3 of them as parameters
3. Compare dates with `isBefore()` and `isAfter()` method calls.